tests/unit/test_network_auth.py

```diff
@@ -47,12 +47,12 @@ def test_get_credentials_parses_correctly(input_url, url, username, password):
     )
 
 
-def test_get_credentials_uses_cached_credentials():
+def test_get_credentials_not_to_uses_cached_credentials():
```

Comment:
Since this test does not test the auth cache anymore, we need to add tests for

1. The cache should be accessed if `_get_new_credentials()` returns a None username and password.
2. *Only* None values from `_get_new_credentials()` are overriden by the credential cache, and username is kept if not None (and the same for password).

Reply:
I've submitted additional tests and small refact as suggested. To my understanding :
(1.) is tested by `test_get_credentials_uses_cached_credentials`
(2.) is tested by `test_get_credentials_not_to_uses_cached_credentials`, `test_get_credentials_not_to_uses_cached_credentials_only_username` and  `test_get_credentials_uses_cached_credentials`